library/ssl_tls13_client.c

```diff
@@ -2649,10 +2652,8 @@ static int ssl_encrypted_extensions_parse( mbedtls_ssl_context* ssl,
 
     while( ext_len )
     {
-        unsigned int ext_id = ( ( ext[0] << 8 )
-                                | ( ext[1] ) );
-        unsigned int ext_size = ( ( ext[2] << 8 )
-                                  | ( ext[3] ) );
+        unsigned int ext_id = ( ( ext[0] << 8 ) | ( ext[1] ) );
```

Comment:
The common idiom is to explicitly cast `buf[x]` to the unsigned target type to avoid implicit promotion to `int`  as part of the shift.